Finished Enemy Action Cycles and Attacks

royjpark opened the pull request:
- Added a boomerang attack for each direction that Goriya faces
- Added the fireball attack on Boss
- Added action cycles for each enemy class, so that when called, it will go through each action.

royjpark committed:
not sure

royjpark committed:
merge

royjpark committed:
merge

royjpark committed:
suo?

royjpark committed:
Implemented bat movement cycle + bat moving nw, ne, sw, and se.

royjpark committed:
Made movement cycle for all enemies.

royjpark committed:
commit again?

royjpark committed:
merge again

royjpark committed:
suo merge

royjpark committed:
merge again ig

royjpark committed:
aye

royjpark committed:
suo again

royjpark committed:
merged w master

royjpark committed:
merge

royjpark committed:
Photoshopped sprite sheets and added boss attack

royjpark committed:
Goriya boomerang started

royjpark committed:
irrelevant

royjpark committed:
irrelevant

royjpark committed:
Merge branch 'master' of https://github.com/sweeneyde/3902-zelda-game into master

royjpark committed:
pull

royjpark committed:
merge

royjpark committed:
irrelevant

royjpark committed:
merge w master

royjpark committed:
irrelevant

royjpark committed:
merged

royjpark committed:
Completed Goriya attacks, and completed cycle of actions for each enemy.

royjpark committed:
Fixed Old Man sprite sheet and changed some var names in BossAttack

bhasseman commented on CrossPlatformDesktopProject/CrossPlatformDesktopProject/IBatState.cs:
```diff
@@ -0,0 +1,16 @@
+﻿using System;
```
From what I could tell, this interface is not implemented anywhere and is no longer needed.

bhasseman commented on CrossPlatformDesktopProject/CrossPlatformDesktopProject/NPC/Bat/Bat.cs:
```diff
@@ -1,18 +1,20 @@
     class Bat : INpc
     {
-        private INpcState currentState;
+        public INpcState currentState;
```
Does this need to be public? It would be best if we can reduce coupling and keep it private, but if it has to be public then it is fine to leave as is for now

sweeneyde replied:
I think it has to remain public so that we can do things like `bat.currentState = new BatWalkSouth(bat);` in `BatWalkEast.cs`.

royjpark replied:
Yes, for now anyways. I need it to be public so that it can go to the next cycle in the animation cycle. So for example within my `BatWalksEast` class, if the bat gets to a certain x position the `bat.CurrentState` will change to the 'BatWalksSouth' class. I need it to do that for sprint 2 just so that each enemy can show off multiple actions when cycled on to

bhasseman commented on CrossPlatformDesktopProject/CrossPlatformDesktopProject/NPC/Boss/Boss.cs:
```diff
@@ -12,7 +12,7 @@ namespace CrossPlatformDesktopProject.NPC
     class Boss : INpc
     {
-        private INpcState currentState;
+        public INpcState currentState;
```
Same with Bat, as well as with the `xPos` and `yPos` from line 16 of this file

royjpark replied:
Yes, same as with Bat I need those to be public

bhasseman reviewed (commented):
Very good work, tested each NPC individually on my end and all compiled and moved as expected. The boss attack looks great. Should be ready to merge once the IBatState file is removed (unless it is needed for some reason that I missed), my other two reviews are not big problems at the moment.

sweeneyde committed:
Merge branch 'master' into NewEnemyAndItemClasses

sweeneyde commented on CrossPlatformDesktopProject/CrossPlatformDesktopProject/NPC/Bat/BatWalkSouth.cs:
```diff
@@ -36,14 +36,18 @@ public void Draw(SpriteBatch spriteBatch, float xPos, float yPos)
 
         public void Update()
         {
+            if (bat.xPos == 450 && bat.yPos == 150)
```
These statments seem awfully hard-coded; maybe keeping track of how long to go rather than where to stop would be better. But we can fix that later.

royjpark replied:
Yeah they're pretty ugly and I was trying to get it to work without hard code but it wasn't going too well, I'll continue to look at it today though.

sweeneyde committed:
Add buttons to change NPCs

sweeneyde commented:
These enemies animations look awesome, nice work!

LGTM

royjpark committed:
irrelevant

royjpark committed:
pull

royjpark commented:
> Very good work, tested each NPC individually on my end and all compiled and moved as expected. The boss attack looks great. Should be ready to merge once the IBatState file is removed (unless it is needed for some reason that I missed), my other two reviews are not big problems at the moment.

I'm pretty sure that I deleted IBatState, and I can't find it anywhere locally. I'm going to push some new commits so could you double check to see if it's still there or not on your end?

royjpark committed:
Minor changes, addressed PR comments

rustbeltdurt committed:
Acceptable amount of Obstacles added.

royjpark committed:
slight changes to world item rectangles

royjpark committed:
made fireball and boomerang starting position less hard coded

rustbeltdurt committed:
Merge pull request #19 from sweeneyde/NewObstacles

Acceptable amount of Obstacles added.

royjpark committed:
Delete IBatState.cs

rustbeltdurt commented:
Pulled down the latest commits and everything looks great. New enemy projectiles look much better. Good work, Roy.
